code/modules/mob_spawn/ghost_roles/pirate_roles.dm

```diff
@@ -21,8 +21,8 @@
 
 /obj/effect/mob_spawn/ghost_role/human/pirate/special(mob/living/spawned_mob, mob/mob_possessor)
 	. = ..()
-	mob_possessor.fully_replace_character_name(mob_possessor.real_name, generate_pirate_name(mob_possessor.gender))
-	mob_possessor.mind.add_antag_datum(/datum/antagonist/pirate)
+	spawned_mob.fully_replace_character_name(spawned_mob.real_name, generate_pirate_name(spawned_mob.gender))
+	spawned_mob.mind.add_antag_datum(/datum/antagonist/pirate)
```

Comment:
is `spawned_mob.mind` guaranteed here? the parent call implies `spawned_mob.mind` can be null

Reply:
@MrMelbert it's only not guaranteed because of fucking golems, it should be cool here

Reply:
works for me